recipes/yaml-cpp/all/test_package/CMakeLists.txt

```
@@ -2,8 +2,10 @@ cmake_minimum_required(VERSION 3.1)
 project(test_package)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
+
+find_package(yaml-cpp REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME} test_package.cpp)
-target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+target_link_libraries(${PROJECT_NAME} yaml-cpp::yaml-cpp)
```

You're creating a `yaml-cpp` aliased target in the main recipe.
```suggestion
target_link_libraries(${PROJECT_NAME} yaml-cpp)
```